### PR TITLE
added small section to PHP Open API describing `PhpFilePathUtils` utility class

### PR DESCRIPTION
**Pull request by i582**

### CODE REVIEW

File: topics/intro/content_updates.md [hunk 1]
```diff
@@ -22,6 +22,7 @@ Inlay Hints
 
 Minor Changes and Additions
 :
+- Add small section to [](php_open_api.md#utility-classes) describing `com.jetbrains.php.lang.psi.elements.impl.PhpFilePathUtils` utility class.
```
Comment (YannCebron) [inline topics/intro/content_updates.md:25]: minor: just use simple class name here

File: topics/products/phpstorm/php_open_api.md [hunk 2]
```diff
@@ -18,6 +18,36 @@
 
 ## Utility Classes
 
+### `com.jetbrains.php.lang.psi.elements.impl.PhpFilePathUtils`
```
Comment (YannCebron) [inline topics/products/phpstorm/php_open_api.md:21]: I'd use simple class name here and clarify FQN in first sentence

File: topics/products/phpstorm/php_open_api.md [hunk 3]
```diff
@@ -18,6 +18,36 @@
...
+
+`PhpFilePathUtils` contains helper methods for working with paths from PSI elements.
+
+#### Useful methods
```
Comment (YannCebron) [inline topics/products/phpstorm/php_open_api.md:25]: not sure we need this heading

File: topics/products/phpstorm/php_open_api.md [hunk 4]
```diff
@@ -18,6 +18,36 @@
...
+
+- `getFileName()` method returns a constant string representation of the path from the PSI element.
```
Comment (YannCebron) [inline topics/products/phpstorm/php_open_api.md:27]: please avoid using terms like "method", "class" etc. in text - it should be clear from notation or context

File: topics/products/phpstorm/php_open_api.md [hunk 5]
```diff
@@ -18,6 +18,36 @@
...
+
+    For example, for the expression:
+
+    ```php
+    // in file: /bin/folder/file.php
+    __DIR__ . "/file2.php";
+    ```
+
+    `getFileName()` will return `"/bin/folder/file2.php"`.
```
Comment (YannCebron) [inline topics/products/phpstorm/php_open_api.md:36]: it returns ...

File: topics/products/phpstorm/php_open_api.md [hunk 6]
```diff
@@ -18,6 +18,36 @@
...
+
+    > The element passed as an argument can contain any expression that can be statically evaluated to a constant.
+    >
+    {type="note"}
```
Comment (YannCebron) [inline topics/products/phpstorm/php_open_api.md:40]: do we really need note element?

File: topics/products/phpstorm/php_open_api.md [hunk 7]
```diff
@@ -18,6 +18,36 @@
...
+
+- `getReferences()` method returns all path references from the PSI element.
+
+    Using this method in conjunction with `com.intellij.psi.PsiReferenceContributor` you can add autocompletion for string literals in the specific contexts.
+    For example, if certain PHP functions in your code accept paths, you can autocomplete them when writing arguments.
+
+    > PhpStorm automatically adds references for concatenation expressions with `__DIR__` or `dirname(__FILE__)` function call.
+    >
+    {type="note"}
```
Comment (YannCebron) [inline topics/products/phpstorm/php_open_api.md:49]: same as above

File: topics/products/phpstorm/php_open_api.md [hunk 8]
```diff
@@ -18,6 +18,36 @@
...
+
+    `getFileName()` will return `"/bin/folder/file2.php"`.
+
+    > The element passed as an argument can contain any expression that can be statically evaluated to a constant.
+    >
+    {type="note"}
+
+- `getReferences()` method returns all path references from the PSI element.
+
+    Using this method in conjunction with `com.intellij.psi.PsiReferenceContributor` you can add autocompletion for string literals in the specific contexts.
```
Comment (YannCebron) [inline topics/products/phpstorm/php_open_api.md:44]: `[`PsiReferenceContributor`](upsource:///platform/core-api/src/com/intellij/psi/PsiReferenceContributor.java)`

File: topics/intro/content_updates.md (hunk 1 above)
Comment (i582) [reply topics/intro/content_updates.md]: Fixed.

File: topics/products/phpstorm/php_open_api.md (hunk 2 above)
Comment (i582) [reply topics/products/phpstorm/php_open_api.md]: Fixed.

File: topics/products/phpstorm/php_open_api.md (hunk 3 above)
Comment (i582) [reply topics/products/phpstorm/php_open_api.md]: Removed.

File: topics/products/phpstorm/php_open_api.md (hunk 4 above)
Comment (i582) [reply topics/products/phpstorm/php_open_api.md]: Fixed.

File: topics/products/phpstorm/php_open_api.md (hunk 5 above)
Comment (i582) [reply topics/products/phpstorm/php_open_api.md]: Fixed.

File: topics/products/phpstorm/php_open_api.md (hunk 6 above)
Comment (i582) [reply topics/products/phpstorm/php_open_api.md]: Fixed.

File: topics/products/phpstorm/php_open_api.md (hunk 7 above)
Comment (i582) [reply topics/products/phpstorm/php_open_api.md]: Fixed.

File: topics/products/phpstorm/php_open_api.md (hunk 8 above)
Comment (i582) [reply topics/products/phpstorm/php_open_api.md]: Fixed.

**Review by YannCebron — APPROVED** [review]

**Comment by YannCebron** [comment]
Thanks!

**Pull request merged**